model/sharing/sharing.go

```diff
@@ -1056,206 +1058,592 @@ func CheckSharings(inst *instance.Instance) ([]map[string]interface{}, error) {
 				})
 			}
-			if s.Triggers.ReplicateID != "" {
+			if s.Credentials[i-1].AccessToken == nil {
 				checks = append(checks, map[string]interface{}{
-					"id":         s.SID,
-					"type":       "trigger_on_inactive_sharing",
-					"trigger":    "replicate",
-					"trigger_id": s.Triggers.ReplicateID,
+					"id":     s.SID,
+					"type":   "missing_access_token",
+					"member": i,
+					"owner":  true,
 				})
 			}
-			if s.Triggers.UploadID != "" {
+			if m.Instance == "" {
 				checks = append(checks, map[string]interface{}{
-					"id":         s.SID,
-					"type":       "trigger_on_inactive_sharing",
-					"trigger":    "upload",
-					"trigger_id": s.Triggers.UploadID,
+					"id":     s.SID,
+					"type":   "missing_instance_for_member",
+					"member": i,
 				})
 			}
 		}
 
-		// Check members and credentials
-		if len(s.Members) < 2 {
+		if len(s.Credentials)+1 != len(s.Members) {
 			checks = append(checks, map[string]interface{}{
 				"id":         s.SID,
-				"type":       "not_enough_members",
-				"nb_members": len(s.Members),
+				"type":       "invalid_number_of_credentials",
+				"owner":      true,
+				"nb_members": len(s.Credentials),
 			})
-			return nil
+			return checks
+		}
+	} else {
+		if len(s.Credentials) != 1 {
+			checks = append(checks, map[string]interface{}{
+				"id":        s.SID,
+				"type":      "invalid_number_of_credentials",
+				"owner":     false,
+				"nb_embers": len(s.Credentials),
+			})
+			return checks
 		}
 
-		for i, m := range s.Members {
-			if m.Status == MemberStatusRevoked && !s.Active {
-				continue
-			}
-			isFirst := i == 0
-			isOwner := m.Status == MemberStatusOwner
-			if isFirst != isOwner {
-				checks = append(checks, map[string]interface{}{
-					"id":     s.SID,
-					"type":   "invalid_member_status",
-					"member": i,
-					"status": m.Status,
-				})
-			}
+		if s.Credentials[0].InboundClientID == "" {
+			checks = append(checks, map[string]interface{}{
+				"id":    s.SID,
+				"type":  "missing_inbound_client_id",
+				"owner": false,
+			})
 		}
 
-		if !s.Active {
-			return nil
+		if !s.ReadOnly() && s.Members[0].Instance == "" {
+			checks = append(checks, map[string]interface{}{
+				"id":     s.SID,
+				"type":   "missing_instance_for_member",
+				"member": 0,
+			})
 		}
+	}
 
-		if s.Owner {
-			for i, m := range s.Members {
-				if i == 0 || m.Status != MemberStatusReady {
-					continue
-				}
-				if s.Credentials[i-1].Client == nil {
+	return checks
+}
+
+func (s *Sharing) checkSharingTreesConsistency(inst *instance.Instance, ownerDocs []couchdb.JSONDoc, freeOwnerDiskSpace int64, m *instance.Instance, ms *Sharing) (checks []map[string]interface{}) {
+	// We checked earlier that this rule exists
+	ownerRule := s.FirstFilesRule()
+
+	memberRule := ms.FirstFilesRule()
+	if memberRule == nil {
+		checks = append(checks, map[string]interface{}{
+			"id":     s.SID,
+			"type":   "missing_files_rule_for_member",
+			"member": m.Domain,
+		})
+		return nil
```

Comment:
It is weird to return nil here, just after assigning to `checks`

Reply:
You're right. It's a mistake introduced when moving around some checks. This should return `checks`.